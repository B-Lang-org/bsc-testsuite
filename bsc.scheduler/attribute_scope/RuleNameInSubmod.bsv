module mkSub (Reg#(UInt#(16)) rg, Empty ifc);
   (* descending_urgency = "sub_r, r" *)
   rule sub_r;
      rg <= rg + 2;
   endrule
endmodule

(* synthesize *)
module sysRuleNameInSubmod ();
   Reg#(UInt#(16)) rg <- mkRegU;

   rule r;
      rg <= rg + 1;
   endrule

   Empty s <- mkSub(rg);
endmodule

