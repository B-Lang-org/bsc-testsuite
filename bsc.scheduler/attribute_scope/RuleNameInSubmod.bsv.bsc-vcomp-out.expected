checking package dependencies
compiling RuleNameInSubmod.bsv
code generation for sysRuleNameInSubmod starts
Error: "RuleNameInSubmod.bsv", line 2, column 36: (G0054)
  Rule `s_r' was referenced in a rule attribute before it was defined.
